BullsAndCowsTest/BullsAndCowsGameTest.cs

```diff
@@ -8,10 +9,106 @@ public class BullsAndCowsGameTest
+        {
+            //given
+            var mockSecretGenerator = new Mock<SecretGenerator>();
+            mockSecretGenerator.Setup(mock => mock.GenerateSecret()).Returns(secrete);
+            var game = new BullsAndCowsGame(mockSecretGenerator.Object);
+            //when
+            string answer = game.Guess(guess);
+            //then
+            Assert.Equal("4A0B", answer);
+        }
+
+        [Theory]
+        [InlineData("4 3 2 1")]
+        [InlineData("3 4 2 1")]
+        [InlineData("4 3 1 2")]
+        public void Should_Return_0A4B_when_all_digit_correct_all_position_wrong(string guess)
+        {
+            //given
+            var testSecretGenerator = new TestSecretGenerator();
+            var game = new BullsAndCowsGame(testSecretGenerator);
+            //when
+            string answer = game.Guess(guess);
+            //then
+            Assert.Equal("0A4B", answer);
+        }
+
+        [Theory]
+        [InlineData("1 4 3 2", "1234")]
+        [InlineData("5 8 7 6", "5678")]
+        public void Should_Return_2A2B_when_all_digit_correct_position_partly_wrong(string guess, string secrete)
+        {
+            //given
+            var mockSecretGenerator = new Mock<SecretGenerator>();
+            mockSecretGenerator.Setup(mock => mock.GenerateSecret()).Returns(secrete);
+            var game = new BullsAndCowsGame(mockSecretGenerator.Object);
+            //when
+            string answer = game.Guess(guess);
+            //then
+            Assert.Equal("2A2B", answer);
+        }
+
+        [Theory]
+        [InlineData("1 4 3 5", "1234")]
+        [InlineData("5 9 7 6", "5678")]
+        public void Should_Return_2A1B_when_digit_partly_correct_position_partly_wrong(string guess, string secrete)
+        {
+            //given
+            var mockSecretGenerator = new Mock<SecretGenerator>();
+            mockSecretGenerator.Setup(mock => mock.GenerateSecret()).Returns(secrete);
+            var game = new BullsAndCowsGame(mockSecretGenerator.Object);
+            //when
+            string answer = game.Guess(guess);
+            //then
+            Assert.Equal("2A1B", answer);
+        }
+
+        [Theory]
+        [InlineData("7 1 4 8", "1234")]
+        [InlineData("8 9 5 3", "5678")]
+        public void Should_Return_0A2B_when_digit_partly_correct_position_all_wrong(string guess, string secrete)
+        {
+            //given
+            var mockSecretGenerator = new Mock<SecretGenerator>();
+            mockSecretGenerator.Setup(mock => mock.GenerateSecret()).Returns(secrete);
+            var game = new BullsAndCowsGame(mockSecretGenerator.Object);
+            //when
+            string answer = game.Guess(guess);
+            //then
+            Assert.Equal("0A2B", answer);
+        }
+
+        public class TestSecretGenerator : SecretGenerator
+        {
+            public override string GenerateSecret()
+            {
+                return "1234";
+            }
+        }
     }
 }
```

Comment:
test coverage: need to add tests for check Input Valid function.

Reply:
+1